src/util.c

```diff
@@ -987,6 +989,20 @@ int fsyncFileDir(const char *filename) {
     return 0;
 }
 
+ /* free OS pages backed by file */
+int reclaimFilePageCache(int fd, size_t offset, size_t length) {
+#ifdef __linux__
```

Comment:
posix_fadvise is POSIX, why do we need that `#ifdef`?
some platforms may not support it by ignoring it, or do different interpretations of the hint, but why shouldn't we give it?

Reply:
According to https://www.gnu.org/software/gnulib/manual/html_node/posix_005ffadvise.html posix_fadvise may not be supported in many platform :( , at least not supported on my MacBook. To suppress the compilation error and for simplicity, I think regard it as a linux-specific feature is fine at the moment. On the other hand, the direct reclaim and cache issue is also linux-specific :)

Reply:
ohh, i imagined that not being supported means that you can call it and it does nothing.
IIRC (from many years ago) even Linux doesn't implement all the advises, or that for some the implementation was really naive, and not taking the full power of what it can do with the advise.

if it fails to compile, maybe there's a better way to detect where it's present, rather than explicitly rely on `__linux__`?

Reply:
@yossigo PATL

Reply:
I found a less intrusive way to achieve this, and I found it works well in my mac and linux vm. I think it's ready to get reviewed.